docs/api_requests/block-vhost-user.md

```diff
@@ -0,0 +1,189 @@
+# Vhost-user block device
+
+As an alternative to [block device](block-io-engine.md) `Sync` and `Async`
+engines, Firecracker supports a vhost-user block device.
+
+There is a good introduction of how a vhost-user block device works
+in general at [FOSDEM23](https://archive.fosdem.org/2023/schedule/event/sds_vhost_user_blk).
+
+[Vhost-user](https://qemu-project.gitlab.io/qemu/interop/vhost-user.html)
+is a userspace protocol that allows to delegate Virtio queue processing
+to another userspace process on the host, as opposed to performing this task
+within Firecracker's VMM thread.
+
+In the vhost-user architecture, the VMM acts as a vhost-user frontend, and
+it is responsible for connecting to the backend via a Unix domain socket (UDS)
+and sharing sufficient information about the guest memory and Virtio queues.
+The vhost-user backend is supposed to receive all this information and perform
+handling of guests's requests. The UDS socket is only used for control purposes
+and does not participate in the dataplane.
+
+Firecracker only implements a vhost-user frontend. Users are free
+to choose from [existing open source backends](#backends) or implement their
+own.
+
+## Topology
+
+Each vhost-user device connects to its own UDS socket. There is no way for
+multiple devices to share a single socket, as there is no way to differentiate
+messages related to devices.
```

Comment:
that is a design choice of the vhost-user protocol, right? If that is so, we could mention that explicitly.

Reply:
no, this is a property of the vhost-user protocol, as far as I know

Reply:
that's what I meant. Maybe we should clarify that. But not extremely important.

Reply:
done